src/pages/_app.tsx

```diff
@@ -11,10 +11,36 @@ import { Sidebar } from '../components/Sidebar/Sidebar';
 import HttpService from '../utils/HttpService';
 import { ROUTES } from '../constants/constants';
 import { useStore } from '../store/store';
+import { useRouter } from 'next/router';
 
-const init = async () => {
+const init = async pathname => {
   const cookbooks = await HttpService.get(ROUTES.COOKBOOKS);
   useStore.setState({ cookbooks });
+
+  const cookbookId = getCookBookIdIfExists(pathname);
+
+  if (cookbookId) {
+    const cookbook = await HttpService.get(ROUTES.COOKBOOK(cookbookId));
+    useStore.setState({ cookbook });
+  }
+
+  let guides = [];
+
+  cookbooks.forEach(async cookbook => {
+    const guidesInCookbook = await HttpService.get(ROUTES.GUIDES(cookbook._id));
+
+    guides = guides.concat(guidesInCookbook);
+
+    useStore.getState().updateGuides(guides);
+  });
+};
+
+const getCookBookIdIfExists = pathname => {
+  const pathArray = pathname.split('/');
+  let cookbookIndex = pathArray.findIndex(path => path == 'cookbooks');
```

Comment:
I think this can be const instead of let cause you don't modify the actual value

Reply:
+1